packages/patternfly-4/react-core/src/components/Pagination/Navigation.tsx

```diff
@@ -54,6 +56,18 @@ export interface NavigationState {
   userInputPage?: React.ReactText;
 }
 
+// Github issue: 3530
+const calculateInputWidth = (lastPage: number) => {
+  const inputControllerWidth = 38;
+  const digitWidth = 8;
+  const digitNo = String(lastPage).length;
+  if (digitNo < 3) {
+    // the current input width fits 2 digits number
+    return null;
+  }
+  return `${inputControllerWidth + digitWidth * digitNo}px`;
```

Comment:
I feel like there should probably be a CSS solution for this in pf-core, instead of hard coding inputControllerWidth (38px) and digitWidth (8px) here.